src/System.Private.CoreLib/src/System/DateTime.Windows.cs

```diff
@@ -0,0 +1,175 @@
+        {
+            long ticks  = DateToTicks(time.systemTime.Year, time.systemTime.Month, time.systemTime.Day);
+            ticks += TimeToTicks(time.systemTime.Hour, time.systemTime.Minute, time.systemTime.Second);
+            ticks += time.systemTime.Milliseconds * TicksPerMillisecond;
+            ticks += time.hundredNanoSecond;
+            return new DateTime( ((UInt64)(ticks)) | KindUtc);
+        }
+
+        private static unsafe bool SystemSupportsLeapSeconds()
+        {
+            Interop.NtDll.SYSTEM_LEAP_SECOND_INFORMATION slsi = new Interop.NtDll.SYSTEM_LEAP_SECOND_INFORMATION();
+            return Interop.NtDll.NtQuerySystemInformation(
+                                    Interop.NtDll.SystemLeapSecondInformation,
+                                    (void *) &slsi,
+                                    sizeof(Interop.NtDll.SYSTEM_LEAP_SECOND_INFORMATION),
+                                    null) == 0 && slsi.Enabled;
+        }
+
+        // FullSystemTime struct is the SYSTEMTIME struct with extra hundredNanoSecond field to store more precise time.
+        [StructLayout(LayoutKind.Sequential)]
+        internal struct FullSystemTime
+        {
+            internal FullSystemTime(int year, int month, DayOfWeek dayOfWeek, int day, int hour, int minute, int second)
+            {
+                systemTime.Year = (ushort) year;
+                systemTime.Month = (ushort) month;
+                systemTime.DayOfWeek = (ushort) dayOfWeek;
+                systemTime.Day = (ushort) day;
+                systemTime.Hour = (ushort) hour;
+                systemTime.Minute = (ushort) minute;
+                systemTime.Second = (ushort) second;
+                systemTime.Milliseconds = 0;
+                hundredNanoSecond = 0;
+            }
+
+            internal FullSystemTime(long ticks)
+            {
+                DateTime dt = new DateTime(ticks);
+
+                int year, month, day;
+                dt.GetDatePart(out year, out month, out day);
+
+                systemTime.Year = (ushort) year;
+                systemTime.Month = (ushort) month;
+                systemTime.DayOfWeek = (ushort) dt.DayOfWeek;
+                systemTime.Day = (ushort) day;
+                systemTime.Hour = (ushort) dt.Hour;
+                systemTime.Minute = (ushort) dt.Minute;
+                systemTime.Second = (ushort) dt.Second;
+                systemTime.Milliseconds = (ushort) dt.Millisecond;
+                hundredNanoSecond = 0;
+            }
+
+            internal Interop.Kernel32.SYSTEMTIME systemTime;
+            internal long   hundredNanoSecond;
+        };
+
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern bool ValidateSystemTime(ref Interop.Kernel32.SYSTEMTIME time, bool localTime);
+
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern bool SystemFileTimeToSystemTime(long fileTime, out FullSystemTime time);
```

Comment:
Is there a reason why we call some of these things `SystemFileTime`? There is `SYSTEMTIME` and `FILETIME` today, but there is no SystemFileTime today.